examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,451 @@
+                   default=DEFAULT_TURN_COUNT)
+    )
+
+    def create_chat_agent(
+            name: str,
+            llm_config: OpenAIGPTConfig,
+            system_message: str
+    ) -> ChatAgent:
+        """creates a ChatAgent with the given parameters.
+
+        Args:
+            name (str): The name of the agent.
+            llm_config (OpenAIGPTConfig): The LLM configuration for the agent.
+            system_message (str): The system message to guide the agent's LLM.
+
+        Returns:
+            ChatAgent: A configured ChatAgent instance.
+        """
+        return ChatAgent(
+            ChatAgentConfig(
+                llm=llm_config,
+                name=name,
+                system_message=system_message,
+            )
+        )
+
+    # Generate the system message
+    feedback_agent_system_message = generate_feedback_agent_system_message(system_messages, pro_key, con_key)
+    pro_agent = create_chat_agent("Pro",
+                                  pro_agent_config,
+                                  system_messages.messages[pro_key].message +
+                                  DEFAULT_SYSTEM_MESSAGE_ADDITION)
+    con_agent = create_chat_agent("Con",
+                                  con_agent_config,
+                                  system_messages.messages[con_key].message +
+                                  DEFAULT_SYSTEM_MESSAGE_ADDITION)
+    feedback_agent = create_chat_agent("Feedback",
+                                       feedback_agent_config,
+                                       feedback_agent_system_message)
+    logger.info("Pro, Con, and feedback agents created.")
+
+    # Determine user's side and assign user_agent and ai_agent based on user selection
+    agents = {"pro": (pro_agent, con_agent, "Pro", "Con"), "con": (con_agent, pro_agent, "Con", "Pro")}
+    user_agent, ai_agent, user_side, ai_side = agents[side]
+    logger.info(
+        f"Starting debate on topic: {topic_name}, taking the {user_side} side. "
+        f"LLM Delegate: {llm_delegate}"
+    )
+
+    logger.info(f"\n{user_side} Agent ({topic_name}):\n")
+
+    # Determine if the debate is autonomous or the user input for one side
+    if llm_delegate:
+        logger.info("Autonomous Debate Selected")
+        interactive_setting = False
+    else:
+        logger.info("Manual Debate Selected with an AI Agent")
+        interactive_setting = True
+        user_input: str = Prompt.ask(
+            "Your argument (or type 'f' for feedback, 'done' to end):"
+        )
+        user_agent.llm = None  # User message without LLM completion
+        user_agent.user_message = user_input
+
+    # Set up langroid tasks and run the debate
+    user_task = Task(user_agent, interactive=interactive_setting, restart=False)
+    ai_task = Task(ai_agent, interactive=False, single_round=True)
+    user_task.add_sub_task(ai_task)
+    if llm_delegate:
```

Comment:
flipped -- needs `not llm_delegate` (fails with gemini in delegate mode since user_message is None)

Reply:
done